### DISCUSSION TITLE
Cant Get Log.<Level>() Func to work

### DISCUSSION THREAD

**Discussion by ALEXTHENINJA007** [custom-components/pyscript#562] [Q&A] [answered]
Syntax errors in Python Scripts get logged; but manually calling log.<level>() in a script doesn't appear to work at all. 
( I can put this as an issue if needed, but I'm hoping I've just dun-goofed and someone can tell me what ive done wrong)

Home Assistant: Core 2023.12.4
pyscript: [1.5.0]
(Jupyter not installed)

Configuration.yaml includes
```
logger:
  default: debug
  logs:
    custom_components.pyscript: debug
```

peanut.py has the following code (yes, I have tried it in explicitly putting the log level for the script in Configuration.yaml)
```
@service(supports_response="optional")
def loggingTest():
	logger.set_level(**{"custom_components.pyscript.file.peanut.loggingTest": "debug"})
	log.debug("test1")
	return {"Complete": True}
```

![image](https://github.com/custom-components/pyscript/assets/1555247/9c6c834c-5cce-4759-acda-f521a9b55a98)

![image](https://github.com/custom-components/pyscript/assets/1555247/5f70ccc8-f946-454b-a1ad-a7d299bae7a4)

**Comment by craigbarratt** [comment, marked as answer]
First, I'm not sure whether the web logs display is the same as the HASS log file.  Can anyone confirm that?

Next, I'd like to confirm that you don't see `log.debug()` messages when you specify:
```yaml
logger:
  default: debug
  logs:
    custom_components.pyscript: debug
```
Did you restart HASS after updating the yaml file?

The dynamic setting in the function creates a race condition.  Since it's implemented as a service call, the `log.debug()` immediately following won't appear since the service call is unlikely to have been executed yet.  I should add that to the documentation....

**Reply by ALEXTHENINJA007** [reply]
I did some checking and confirm it gets logged to the _home-assistant.log_ file
